arraycontext/impl/pyopencl/taggable_cl_array.py

```diff
@@ -125,5 +122,8 @@ def to_tagged_cl_array(ary: cla.Array,
         array. If passed *None*, then initialized to a :class:`pytato.Axis`
         with no tags attached for each dimension.
     """
+    if isinstance(ary, TaggableCLArray):
+        return ary
+
```

Comment:
This slightly worries me that the tags passed to `to_tagged_cl_array` are ignored. OTOH, I see your point of how do we handle conflicts. How about we preserve both the tags (the ones present in `ary.tags` and those passed in as `tags` arguments) and clarify this behavior in the docs?

Reply:
Oh, yeah, sorry, I should mark this as a draft still. 

I'll look into transfering the tags.

Reply:
Handled the tags in `to_tagged_cl_array` in 2ac228f. Not super sure the `axes` are handled correctly, so let me know if anything is fishy there!